src/lib.rs

```diff
@@ -219,6 +219,18 @@ unsafe impl<'a> Alloc for &'a Jemalloc {
     }
 }
 
+/// Return the usable size of the allocation pointed to by ptr.
+///
+/// The return value may be larger than the size that was requested during allocation.
+/// This function is not a mechanism for in-place `realloc()`;
+/// rather it is provided solely as a tool for introspection purposes.
+/// Any discrepancy between the requested allocation size
+/// and the size reported by this function should not be depended on,
+/// since such behavior is entirely implementation-dependent.
+pub unsafe fn usable_size<T: Sized>(ptr: *const T) -> usize {
```

Comment:
Is `: Sized` needed here?

Reply:
I guess it’s not in the sense that it’s the default, but `?Sized` would cause the cast to `*const c_void` to not work… won’t it? Hmm, I just tried and it looks like an casting unsized raw pointer to sized is allowed, though documenting it is "TODO" in https://doc.rust-lang.org/nightly/nomicon/casts.html. I don’t know if it’s desirable here, though.

Reply:
Oh sorry I just mean that `: Sized` is the default, so the choices here are basically `: ?Sized` or no bounds at all, but I like the idea of using a sized pointer as jemalloc isn't the one allocating vtables.